otelx/otlp.go

```diff
@@ -21,22 +22,15 @@ import (
 func SetupOTLP(t *Tracer, tracerName string, c *Config) (trace.Tracer, error) {
 	ctx := context.Background()
 
-	clientOpts := []otlptracehttp.Option{
-		otlptracehttp.WithEndpoint(c.Providers.OTLP.ServerURL),
-	}
-
-	if c.Providers.OTLP.Insecure {
-		clientOpts = append(clientOpts, otlptracehttp.WithInsecure())
-	}
-
-	if c.Providers.OTLP.AuthorizationHeader != "" {
-		clientOpts = append(clientOpts,
-			otlptracehttp.WithHeaders(map[string]string{"Authorization": c.Providers.OTLP.AuthorizationHeader}),
-		)
+	var client otlptrace.Client
+	if c.Providers.OTLP.Transport == "GRPC" {
+		client = getOTLPGRPClient(c)
+	} else {
```

Comment:
It may be required to add `http/json` and `http/protobuf` in the future. At the moment we default to the current behaviour
https://opentelemetry.io/docs/specs/otel/protocol/exporter/#specify-protocol